client/src/app.ts

```diff
@@ -0,0 +1,121 @@
+import {UserAction , Role} from "./models/model.js";
+
+function userCreatedDate(){
+    return function(target: any,key:string,descriptor:PropertyDescriptor){
+        let val = descriptor.value
+        descriptor.value = function(... args:any[]){
+            let arr:Array<string> = [];
+            for(let j=0;j<7;j++){
```

Comment:
try to use descriptive constants instead of numbers. That makes code more readable.

Reply:
changed the numbers into more readable format